scripts/rewrite_app_version_number.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+# [Script run by publish-extension workflow]
+# Replaces the version numbers in app.js and service-worker.js with the following priority:
+#
+# 1. use the override value set in the workflow dispatch;
+# 2. use the tag pattern (if no override value was set)
+#
+# This script can be tested manually by replacing ${INPUT_VERSION} and ${TAG_VERSION}
+# with test strings
+ 
+# Use the override value by preference
+VERSION=${INPUT_VERSION}
+if [[ $VERSION =~ ^v*[0-9.]+ ]]; then
```

Comment:
I agree, and I accept that change. The reason I'd kept the star is because that's the version given in the GitHub workflow as the tag-pattern. It's the default one given in the documentation, but I actually did thing that it was strange that it could match `vvvvvvvvv3.1.1' . If we change it here, we should change in the workflow too.